src/app/components/compact-list/compact-list.component.ts

```diff
@@ -0,0 +1,184 @@
+import {Component, Input, Output, ElementRef, Renderer, ViewChild} from "@angular/core";
+import {FormGroup, FormControl, Validators} from "@angular/forms";
+import {ReplaySubject} from "rxjs";
+import {ComponentBase} from "../common/component-base";
+import {ContenteditableDirective} from "./directives/contenteditable.directive";
+import {TagModel} from "./tag.model";
+
+require("./compact-list.component.scss");
+
+export interface ValidationResponse {
+    isValid: boolean,
+    errorMessage: string
+}
+
+@Component({
+    selector: "compact-list",
+    directives: [
+        ContenteditableDirective
+    ],
+    template: `
+       <form [formGroup]="formGroup" class="compact-list-form">
+            <div class="compact-list-wrapper" (click)="onListWrapperClick()">
+                
+                <div class="input-tag-list">
+                    <span *ngFor="let tag of tagList; let i = index; trackBy:trackByIndex"
+                          class="tag tag-pill tag-default">
+                          
+                          <span *ngIf="tag.validation?.errors?.length > 0"
+                                title="tag.validation.errorText"
+                                class="icon-wrapper">
+                                <i class="fa fa-times-circle error">
+                                </i>
+                          </span>
+                          
+                          <span *ngIf="tag.validation?.warnings?.length > 0"
+                                class="icon-wrapper">
+                                <i class="fa fa-warning warning">
+                                </i>
+                          </span>
+                          
+                          {{tag.value}}
+                          <i class="fa fa-times remove-tag-icon"
+                             (click)="removeTag(i, $event)"></i>
+                    </span>
+                   
+                   
+                    <!-- Not using the <input>, 
+                         so that the width can adjust to the text length,
+                         and break into a new line if its long -->
+                    <span #tagInput 
+                          contenteditable="true"
+                          class="tag-input"
+                          *ngIf="formGroup.controls['compactListControl']"
+                          [ngClass]="{'invalid-input': !isValidInput }"
+                          [formControl]="formGroup.controls['compactListControl']"
```

Comment:
This approach is inflexible. It expects there to be only one compact-list component per formGroup.